app/geant-exporter/GeantExceptionHandler.cc

```diff
@@ -0,0 +1,55 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file GeantExceptionHandler.cc
+//---------------------------------------------------------------------------//
+#include "GeantExceptionHandler.hh"
+
+#include <sstream>
+#include "base/Assert.hh"
+#include "comm/Logger.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Propagate exceptions to Celeritas.
+ */
+G4bool GeantExceptionHandler::Notify(const char*         origin_of_exception,
+                                     const char*         exception_code,
+                                     G4ExceptionSeverity severity,
+                                     const char*         description)
+{
+    REQUIRE(origin_of_exception);
+    REQUIRE(exception_code);
+
+    // Construct message
+    std::ostringstream os;
+    os << exception_code << " in Geant4 " << origin_of_exception << ": "
+       << description;
+    const std::string& msg = os.str();
+
+    switch (severity)
+    {
+        case FatalException:
+        case FatalErrorInArgument:
+        case RunMustBeAborted:
+        case EventMustBeAborted:
+            // Severe or initialization error
+            throw celeritas::DebugError(msg);
```

Comment:
The name 'DebugError' is a bit funny sounding :).  Does it really say what you mean?

Reply:
You're right, `DebugError` is a `std::logic_error` and this isn't for debugging. I'll make a `RuntimeError` class that inherits from `std::runtime_error` for the "user-facing assertion" cases like this.